src/details/ArborX_Ray.hpp

```diff
@@ -289,6 +296,184 @@ KOKKOS_INLINE_FUNCTION float overlapDistance(Ray const &ray,
   return (tmax - tmin);
 }
 
+KOKKOS_INLINE_FUNCTION Quaternion hamiltonProduct(Quaternion const &a,
+                                                  Quaternion const &b)
+{
+  Quaternion res;
+  res.r = a.r * b.r - a.i * b.i - a.j * b.j - a.k * b.k;
+  res.i = a.r * b.i + a.i * b.r + a.j * b.k - a.k * b.j;
+  res.j = a.r * b.j - a.i * b.k + a.j * b.r + a.k * b.i;
+  res.k = a.r * b.k + a.i * b.j - a.j * b.i + a.k * b.r;
+
+  return res;
+}
+
+KOKKOS_INLINE_FUNCTION Point rotatePoint(Point const &point,
+                                         Quaternion const &rotation_quat)
+{
+  Quaternion p = {0, point[0], point[1], point[2]};
+  Quaternion rotation_quat_inv = {rotation_quat.r, -rotation_quat.i,
+                                  -rotation_quat.j, -rotation_quat.k};
+  auto rotated_p =
+      hamiltonProduct(hamiltonProduct(rotation_quat, p), rotation_quat_inv);
+
+  return {rotated_p.i, rotated_p.j, rotated_p.k};
+}
+
+KOKKOS_INLINE_FUNCTION Quaternion computeQuaternionRotation(float const &angle,
+                                                            Vector const &axis)
+{
+  Quaternion rotation;
+  rotation.r = std::cos(angle / 2.);
```

Comment:
Did you intend the cast to double-precision floating-point?  If not `/ 2. -> / 2`

Reply:
I did want `double` when I was debugging but I think `float` should work too.